yarn-install/action.yml

```
@@ -23,7 +30,18 @@ runs:
         # We use both yarn.lock and package.json as cache keys to ensure that
         # changes to local monorepo packages bust the cache.
         key: ${{ inputs.cache-prefix }}-node_modules-${{ hashFiles('yarn.lock', '**/package.json') }}
-
+    
+    - name: prepare corepack
+      shell: bash
+      run: |
+        if [[ "${{ inputs.yarn-install-force }}" == "true" ]]; then
+          npm install --force -g corepack@${{ inputs.corepack-version }}
+        else
+          npm install -g corepack@${{ inputs.corepack-version }}
+        fi
```

This could be simplified a bit.

```suggestion
      env:
        YARN_INSTALL_FORCE: ${{ inputs.yarn-install-force == 'true' && '--force' || '' }}
      run: |
        npm install $YARN_INSTALL_FORCE -g corepack@${{ inputs.corepack-version }}
```